<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>fen_enreg_utilisateur</class>
 <widget class="QDialog" name="fen_enreg_utilisateur">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>383</width>
    <height>383</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>383</width>
    <height>383</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <property name="styleSheet">
   <string notr="true">
*{

		/*border: 1px solid rgb(133, 133, 133);*/
		
		color: rgb(74, 74, 74);
		font: 350 10pt &quot;Bahnschrift&quot;;

}


QDialog , #fen_enreg_utilisateur{

		
			border-radius :10px ;

}

QGroupBox{

		border:none ;

}

#label_affiche_erreur_validation{
		
		
	font: italic 8pt &quot;Segoe UI&quot;;

	
	color: rgb(218, 17, 54);

}

#label_description_fenetre{

	
	font: 300 8pt &quot;Bahnschrift&quot;;
	
	color: rgb(113, 113, 113);

}

#label_titre_fenetre{
	
		
	font: 600 12pt &quot;Bahnschrift&quot;;

}


#fen_enreg_utilisateur{

		background-color: rgb(255, 255, 255);
    border-radius: 4px;
    border: 1px solid rgb(229, 234, 239);
}


/*===============================================
   BOUTONS STANDARDS
   ===============================================*/


QPushButton {
    background-color: rgb(13, 110, 253);
    padding: 6px 12px;
    border-radius: 4px;
    border: none;
    color: rgb(255, 255, 255);
    font-weight: 500;
    min-height: 18px;
}

QPushButton:hover {
    background-color: rgb(10, 88, 202);
}

QPushButton:pressed {
    background-color: rgb(8, 77, 178);
}

QToolButton {
    background: transparent;
    border: none;
    outline: none;
}


#btn_annuler_ajout_utilisateur{

		border:none ;
	   color: rgb(64, 64, 64);
		background : transparent ;

}

#btn_annuler_ajout_utilisateur:hover{

	color:rgb(13, 110, 253);

}

/*===============================================
   ÉLÉMENTS DE FORMULAIRE
   ===============================================*/

QLineEdit, QTextEdit {
    background-color: rgb(255, 255, 255);
    border: 1px solid #ced4da;
    border-radius: 4px;
    padding: 6px;
    color: #495057;
}

QLineEdit:hover, QTextEdit:hover {
    border: 1px solid #0d6efd;
    background-color: rgb(248, 249, 250);
}

QLineEdit:focus, QTextEdit:focus, 
QComboBox:focus, QDoubleSpinBox:focus {
    border: 1px solid #0d6efd;
    outline: none;
}

/*===============================================
   COMBOBOX
   ===============================================*/

QComboBox {
    background-color: rgb(255, 255, 255);
    border: 1px solid #ced4da;
    border-radius: 4px;
    padding: 6px 24px 6px 8px;
    color: #495057;
    font-size: 13px;
    min-height: 18px;
    selection-background-color: #0d6efd;
}

QComboBox:hover {
    border-color: #0d6efd;
    background-color: rgb(248, 249, 250);
}

QComboBox:focus {
    border: 1px solid #0d6efd;
    outline: none;
}

QComboBox:pressed {
    background-color: rgb(240, 244, 248);
}

QComboBox::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right;
    width: 20px;
    border: none;
    background-color: transparent;
    border-top-right-radius: 4px;
    border-bottom-right-radius: 4px;
}

QComboBox::drop-down:hover {
    background-color: rgba(229, 234, 239, 0.3);
}

QComboBox::drop-down:pressed {
    background-color: rgba(229, 234, 239, 0.5);
}

QComboBox::down-arrow {
    image: url(:/icon_gris/chevron-down.svg);
    width: 12px;
    height: 12px;
}

QComboBox::down-arrow:pressed {
    image: url(:/icon_noir/chevron-down.svg);
}

QComboBox QAbstractItemView {
    border: 1px solid #0d6efd;
    border-radius: 4px;
    background-color: rgb(255, 255, 255);
    selection-background-color: #0d6efd;
    selection-color: white;
    padding: 2px;
    outline: none;
    show-decoration-selected: 1;
}

QComboBox QAbstractItemView::item {
    height: 28px;
    padding: 0 8px;
    border-bottom: 1px solid rgba(229, 234, 239, 0.5);
    color: #495057;
}

QComboBox QAbstractItemView::item:last {
    border-bottom: none;
}

QComboBox QAbstractItemView::item:hover {
    background-color: rgba(13, 110, 253, 0.1);
    color: #0d6efd;
}

QComboBox QAbstractItemView::item:selected {
    background-color: #0d6efd;
    color: white;
}

/*===============================================
   SPINBOX ET DOUBLESPINBOX
   ===============================================*/

QDoubleSpinBox, QSpinBox {
    background-color: rgb(255, 255, 255);
    border: 1px solid #ced4da;
    border-radius: 4px;
    padding: 6px 8px;
    color: #495057;
    font-size: 13px;
    min-height: 18px;
    selection-background-color: #0d6efd;
}

QDoubleSpinBox:hover, QSpinBox:hover {
    border-color: #0d6efd;
    background-color: rgb(248, 249, 250);
}

QDoubleSpinBox:focus, QSpinBox:focus {
    border: 1px solid #0d6efd;
    outline: none;
}

QDoubleSpinBox::up-button, QSpinBox::up-button {
    subcontrol-origin: border;
    subcontrol-position: top right;
    width: 18px;
    height: 10px;
    border: none;
    background-color: transparent;
    margin-top: 1px;
    margin-right: 1px;
}

QDoubleSpinBox::up-button:hover, QSpinBox::up-button:hover {
    background-color: rgba(229, 234, 239, 0.3);
}

QDoubleSpinBox::up-button:pressed, QSpinBox::up-button:pressed {
    background-color: rgba(229, 234, 239, 0.5);
}

QDoubleSpinBox::down-button, QSpinBox::down-button {
    subcontrol-origin: border;
    subcontrol-position: bottom right;
    width: 18px;
    height: 10px;
    border: none;
    background-color: transparent;
    margin-bottom: 1px;
    margin-right: 1px;
}

QDoubleSpinBox::down-button:hover, QSpinBox::down-button:hover {
    background-color: rgba(229, 234, 239, 0.3);
}

QDoubleSpinBox::down-button:pressed, QSpinBox::down-button:pressed {
    background-color: rgba(229, 234, 239, 0.5);
}

QDoubleSpinBox::up-arrow, QSpinBox::up-arrow {
    image: url(:/icon_gris/chevron-up.svg);
    width: 8px;
    height: 8px;
}

QDoubleSpinBox::up-arrow:hover, QSpinBox::up-arrow:hover {
    image: url(:/icon_noir/chevron-up.svg);
}

QDoubleSpinBox::down-arrow, QSpinBox::down-arrow {
    image: url(:/icon_gris/chevron-down.svg);
    width: 8px;
    height: 8px;
}

QDoubleSpinBox::down-arrow:hover, QSpinBox::down-arrow:hover {
    image: url(:/icon_noir/chevron-down.svg);
}

QDoubleSpinBox::up-arrow:disabled, QSpinBox::up-arrow:disabled {
    image: url(:/icon_gris/chevron-up-disabled.svg);
}

QDoubleSpinBox::down-arrow:disabled, QSpinBox::down-arrow:disabled {
    image: url(:/icon_gris/chevron-down-disabled.svg);
}

/*===============================================
   DATEEDIT
   ===============================================*/

QDateEdit {
    border: 1px solid #ced4da;
    border-radius: 4px;
    padding: 6px 24px 6px 8px;
    background-color: rgb(255, 255, 255);
    color: #495057;
    font-size: 13px;
    min-height: 18px;
}

QDateEdit:hover {
    border-color: #0d6efd;
    background-color: rgb(248, 249, 250);
}

QDateEdit:focus {
    border: 1px solid #0d6efd;
    outline: none;
}

QDateEdit::drop-down {
    subcontrol-origin: padding;
    subcontrol-position: top right;
    width: 20px;
    border: none;
    border-left: 1px solid #ced4da;
    border-top-right-radius: 4px;
    border-bottom-right-radius: 4px;
    background-color: rgba(229, 234, 239, 0.3);
}

QDateEdit::drop-down:hover {
    background-color: rgba(229, 234, 239, 0.6);
}

QDateEdit::down-arrow {
    image: url(:/icon_gris/calendar.svg);
    width: 12px;
    height: 12px;
}

/*===============================================
   ONGLETS DE TRANSACTION
   ===============================================*/

#mes_onglets_page_transaction {
    background-color: rgb(248, 249, 250);
    border: none;
    border-radius: 4px;
}

#mes_onglets_page_transaction::pane {
    border: none;
    border-radius: 0 0 4px 4px;
    background-color: rgb(248, 249, 250);
}

#mes_onglets_page_transaction &gt; QTabBar {
    background: transparent;
    border: none;
    spacing: 4px;
}

#mes_onglets_page_transaction &gt; QTabBar::tab {
    background-color: rgba(229, 234, 239, 0.5);
    border: 1px solid rgb(229, 234, 239);
    border-bottom: none;
    border-radius: 4px 4px 0 0;
    padding: 8px 16px;
    margin-right: 2px;
    min-width: 100px;
    color: rgb(108, 117, 125);
}

#mes_onglets_page_transaction &gt; QTabBar::tab:selected {
    background-color: rgb(248, 249, 250);
    border-color: rgb(229, 234, 239);
    color: rgb(73, 80, 87);
    font-weight: 600;
    border-bottom: 2px solid #0d6efd;
    margin-bottom: -1px;
}

#mes_onglets_page_transaction &gt; QTabBar::tab:hover {
    background-color: rgba(229, 234, 239, 0.8);
    color: rgb(13, 110, 253);
}

#mes_onglets_page_transaction &gt; QTabBar::tab:selected:hover {
    background-color: rgb(233, 236, 239);
}

#mes_onglets_page_transaction &gt; QTabBar QToolButton {
    background: rgb(229, 234, 239);
    border-radius: 3px;
    margin: 4px;
}

#mes_onglets_page_transaction &gt; QTabBar QToolButton::right-arrow,
#mes_onglets_page_transaction &gt; QTabBar QToolButton::left-arrow {
    image: url(:/icon_gris/chevron-right.svg);
    width: 12px;
    height: 12px;
}

/*===============================================
   BARRES DE DÉFILEMENT
   ===============================================*/

QScrollBar:vertical {
    border: none;
    background-color: rgba(248, 249, 250, 0);
    width: 10px;
    margin: 0;
}

QScrollBar:horizontal {
    border: none;
    background-color: rgba(248, 249, 250, 0);
    height: 10px;
    margin: 0;
}

QScrollBar::handle:vertical,
QScrollBar::handle:horizontal {
    background-color: rgba(206, 212, 218, 0.5);
    border-radius: 4px;
    min-height: 20px;
    min-width: 20px;
    opacity: 0;
}

QScrollBar::handle:vertical:hover,
QScrollBar::handle:horizontal:hover {
    background-color: rgba(13, 110, 253, 0.6);
    opacity: 1;
}

QScrollBar::handle:vertical:pressed,
QScrollBar::handle:horizontal:pressed {
    background-color: rgba(8, 77, 178, 0.8);
}

QScrollBar::add-line:vertical,
QScrollBar::sub-line:vertical,
QScrollBar::add-line:horizontal,
QScrollBar::sub-line:horizontal {
    border: none;
    background: none;
    height: 0;
    width: 0;
}

QScrollBar:vertical:hover,
QScrollBar:horizontal:hover {
    background-color: rgba(248, 249, 250, 0.3);
}

QScrollBar::add-page:vertical,
QScrollBar::sub-page:vertical,
QScrollBar::add-page:horizontal,
QScrollBar::sub-page:horizontal {
    background: none;
}

/*===============================================
   SÉPARATEURS
   ===============================================*/

Line {
    background-color: rgb(229, 234, 239);
    border-radius: 1px;
    border: 1px solid rgb(229, 234, 239);
    width: 1px;
}

#line, #line_2 ,#line_3{
    border: none;
    background-color: rgba(206, 212, 218, 0.7);
    border-radius: 1px;
    outline: none;
}

/*===============================================
   TABLEAUX - STYLE AMÉLIORÉ AVEC CELLULES PLUS GRANDES
   ===============================================*/

QTableWidget {
    background-color: #ffffff;
    alternate-background-color: #ffffff;
    gridline-color: #e0e0e0;
    color: #333333;
    border:  1px solid  rgb(224, 224, 224);
    selection-background-color: #e3f2fd;
    selection-color: #1976d2;
    outline: none;
	border-radius : 8px ;
}

QTableWidget QHeaderView::section {
    background-color: #f5f5f5;
    color: #333333;
    padding: 6px 4px;
    border: none;
    border-bottom: 1px solid #e0e0e0;
    font-weight: bold;
    font-size: 9pt;
    text-align: left;
    min-height: 35px;
}

QTableWidget QHeaderView::vertical::section {
    background-color: #ffffff;
    border: none;
    min-width: 45px;
}

QTableWidget::item {
    padding: 6px 4px;
    border-bottom: 1px solid #e0e0e0;
    border-right: none;
    min-height: 40px;
    font-size: 7pt;
}

QTableWidget::item:hover {
    background-color: #f0f0f0;
}

QTableWidget::item:selected {
    background-color: #e3f2fd;
    color: #1976d2;
}

/*===============================================
BADGES DE MONTANT - STYLE INSPIRÉ DES BADGES DE STATUT MODERNES
===============================================*/

QLabel.MontantBadgePositif {
    background-color: rgba(34, 197, 94, 0.2);
    color: #22c55e;
    border: 1px solid rgba(34, 197, 94, 0.3);
    border-radius: 8px;
    padding: 3px 6px;
    min-width: 55px;
    max-width: 105px;
    text-align: center;
    font-weight: 500;
    font-size: 6pt;
}

QLabel.MontantBadgeNegatif {
    background-color: rgba(239, 68, 68, 0.2);
    color: #ef4444;
    border: 1px solid rgba(239, 68, 68, 0.3);
    border-radius: 8px;
    padding: 3px 6px;
    min-width: 55px;
    max-width: 105px;
    text-align: center;
    font-weight: 500;
    font-size: 6pt;
}

QLabel.MontantBadgeNeutre {
    background-color: rgba(107, 114, 128, 0.2);
    color: #6b7280;
    border: 1px solid rgba(107, 114, 128, 0.3);
    border-radius: 8px;
    padding: 3px 6px;
    min-width: 55px;
    max-width: 105px;
    text-align: center;
    font-weight: 500;
    font-size: 6pt;
}

QLabel.MontantBadgePending {
    background-color: rgba(245, 158, 11, 0.2);
    color: #f59e0b;
    border: 1px solid rgba(245, 158, 11, 0.3);
    border-radius: 8px;
    padding: 3px 6px;
    min-width: 55px;
    max-width: 105px;
    text-align: center;
    font-weight: 500;
    font-size: 6pt;
}

QLabel.MontantBadgeCritique {
    background-color: rgba(239, 68, 68, 0.9);
    color: white;
    border: 1px solid #ef4444;
    border-radius: 12px;
    padding: 4px 12px;
    min-width: 60px;
    max-width: 120px;
    text-align: center;
    font-weight: 500;
    font-size: 12px;
}

/*===============================================
VARIANTES AVEC EFFET HOVER (OPTIONNEL)
===============================================*/

QLabel.MontantBadgePositif:hover {
    background-color: rgba(34, 197, 94, 0.3);
    border: 1px solid rgba(34, 197, 94, 0.5);
}

QLabel.MontantBadgeNegatif:hover {
    background-color: rgba(239, 68, 68, 0.3);
    border: 1px solid rgba(239, 68, 68, 0.5);
}

QLabel.MontantBadgeNeutre:hover {
    background-color: rgba(107, 114, 128, 0.3);
    border: 1px solid rgba(107, 114, 128, 0.5);
}

QLabel.MontantBadgePending:hover {
    background-color: rgba(245, 158, 11, 0.3);
    border: 1px solid rgba(245, 158, 11, 0.5);
}


/* Style pour les boutons actifs */
#BarreLaterale QToolButton[active=&quot;true&quot;] {
    background-color: #1a3c6c;
    color: white;
    padding: 4px 4px;
    text-align: left;
    border-radius: 4px;
}

/* Style pour les boutons normaux */
#BarreLaterale QToolButton {
    background-color: transparent;
    border: none;
    color: rgb(73, 80, 87);
    padding: 4px 4px;
    text-align: left;
    border-radius: 4px;
}</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_6">
   <property name="leftMargin">
    <number>4</number>
   </property>
   <property name="topMargin">
    <number>4</number>
   </property>
   <property name="rightMargin">
    <number>4</number>
   </property>
   <property name="bottomMargin">
    <number>4</number>
   </property>
   <item>
    <widget class="QWidget" name="widget" native="true">
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="spacing">
       <number>8</number>
      </property>
      <property name="leftMargin">
       <number>2</number>
      </property>
      <property name="topMargin">
       <number>2</number>
      </property>
      <property name="rightMargin">
       <number>2</number>
      </property>
      <property name="bottomMargin">
       <number>2</number>
      </property>
      <item>
       <widget class="QLabel" name="label_titre_fenetre">
        <property name="text">
         <string>Ajouter un Utilisateur</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="label_description_fenetre">
        <property name="text">
         <string>Créer un nouveau compte utilisateur et assigné un rôle</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="Line" name="line">
     <property name="orientation">
      <enum>Qt::Orientation::Horizontal</enum>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Nom complet</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <property name="spacing">
       <number>20</number>
      </property>
      <property name="leftMargin">
       <number>2</number>
      </property>
      <property name="topMargin">
       <number>15</number>
      </property>
      <property name="rightMargin">
       <number>2</number>
      </property>
      <property name="bottomMargin">
       <number>2</number>
      </property>
      <item>
       <widget class="QLineEdit" name="lineEdit_nom_complet_utilisateur">
        <property name="inputMask">
         <string/>
        </property>
        <property name="placeholderText">
         <string>EX : Samba denos</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Email</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <property name="spacing">
       <number>2</number>
      </property>
      <property name="leftMargin">
       <number>2</number>
      </property>
      <property name="topMargin">
       <number>15</number>
      </property>
      <property name="rightMargin">
       <number>2</number>
      </property>
      <property name="bottomMargin">
       <number>2</number>
      </property>
      <item>
       <widget class="QLineEdit" name="lineEdit_email_utilisateur">
        <property name="placeholderText">
         <string>Xyz@gmail.com</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_4">
     <property name="title">
      <string>Mot de passe </string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_7">
      <property name="spacing">
       <number>2</number>
      </property>
      <property name="leftMargin">
       <number>2</number>
      </property>
      <property name="topMargin">
       <number>15</number>
      </property>
      <property name="rightMargin">
       <number>2</number>
      </property>
      <property name="bottomMargin">
       <number>2</number>
      </property>
      <item>
       <widget class="QLineEdit" name="lineEdit_mot_de_passe">
        <property name="echoMode">
         <enum>QLineEdit::EchoMode::PasswordEchoOnEdit</enum>
        </property>
        <property name="placeholderText">
         <string>Créer votre mot de passe </string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_5">
     <property name="title">
      <string>Confirmation de mot de passe </string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_8">
      <property name="spacing">
       <number>2</number>
      </property>
      <property name="leftMargin">
       <number>2</number>
      </property>
      <property name="topMargin">
       <number>15</number>
      </property>
      <property name="rightMargin">
       <number>2</number>
      </property>
      <property name="bottomMargin">
       <number>2</number>
      </property>
      <item>
       <widget class="QLineEdit" name="lineEdit_confirmation_mot_de_passe">
        <property name="echoMode">
         <enum>QLineEdit::EchoMode::Password</enum>
        </property>
        <property name="placeholderText">
         <string>Confirmer votre mot de passe </string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Rôle </string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_4">
      <property name="spacing">
       <number>20</number>
      </property>
      <property name="leftMargin">
       <number>2</number>
      </property>
      <property name="topMargin">
       <number>15</number>
      </property>
      <property name="rightMargin">
       <number>2</number>
      </property>
      <property name="bottomMargin">
       <number>2</number>
      </property>
      <item>
       <widget class="QComboBox" name="comboBox_role_utilisateur"/>
      </item>
     </layout>
    </widget>
   </item>
   <item alignment="Qt::AlignmentFlag::AlignVCenter">
    <widget class="QLabel" name="label_affiche_erreur_validation">
     <property name="text">
      <string/>
     </property>
     <property name="alignment">
      <set>Qt::AlignmentFlag::AlignCenter</set>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QWidget" name="widget_3" native="true">
     <layout class="QVBoxLayout" name="verticalLayout_5">
      <item alignment="Qt::AlignmentFlag::AlignRight">
       <widget class="QWidget" name="widget_2" native="true">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <property name="spacing">
          <number>20</number>
         </property>
         <property name="leftMargin">
          <number>2</number>
         </property>
         <property name="topMargin">
          <number>2</number>
         </property>
         <property name="rightMargin">
          <number>2</number>
         </property>
         <property name="bottomMargin">
          <number>2</number>
         </property>
         <item>
          <widget class="QPushButton" name="btn_annuler_ajout_utilisateur">
           <property name="cursor">
            <cursorShape>PointingHandCursor</cursorShape>
           </property>
           <property name="text">
            <string>Annuler</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pushButton">
           <property name="cursor">
            <cursorShape>PointingHandCursor</cursorShape>
           </property>
           <property name="text">
            <string>Créer</string>
           </property>
           <property name="icon">
            <iconset resource="icones.qrc">
             <normaloff>:/icon_blanc/plus-circle.svg</normaloff>:/icon_blanc/plus-circle.svg</iconset>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="icones.qrc"/>
 </resources>
 <connections/>
</ui>
